lib/api/core/models/repositories/repository.js

```diff
@@ -252,6 +313,27 @@ Repository.prototype.persistToDatabase = function (object) {
 };
 
 /**
+ * Delete repository from database according to its id
+ * @param id
+ */
+Repository.prototype.deleteFromDatabase = function (id) {
```

Comment:
We've already discussed with @benoitvidis and the conclusion was to let each repo (profile, role, user) to delete from its memory and the meta repository for delete from the database.
the method `deleteRole` delete the role from memory and next call this `deleteFromDatabase`.
@benoitvidis maybe I misunderstood ?